libraries/APM_Control/AR_AttitudeControl.cpp

```diff
@@ -576,15 +576,15 @@ float AR_AttitudeControl::get_sail_out_from_heel(float desired_heel, float dt)
 
     // get p
     float p = _sailboat_heel_pid.get_p();
-    // constrain p to only be positive
-    if (!is_positive(p)) {
+    // constrain p to only be negative
+    if (!is_negative(p)) {
```

Comment:
So the P and I terms of the HEEL PID will be negative.  That cannot be correct.  @lthall maybe we need to discuss the fix after the latest changes?

Reply:
Leonard helped my work this one out, my original error was calculated wrong, the correct method now used gives -1 * the error I was using before. Hence the change from is_positive to is_negative. Then the whole lot is multiplied by -1 at the end so it looks the same to everything else. I have tested this on the boat (on the stand).